src/main/scala/zio/codec/CodecModule.scala

```diff
@@ -57,7 +60,7 @@ trait CodecModule {
     private[zio] sealed case class Fail[A](error: String)                                            extends Codec[A]
     private[zio] case object Consume                                                                 extends Codec[Input]
     private[zio] sealed case class Map[A, B](equiv: Equiv[A, B], value: () => Codec[A])              extends Codec[B]
-    private[zio] sealed case class Filter[A](value: () => Codec[A], filter: Set[A])                  extends Codec[A]
+    private[zio] sealed case class Filter[A](value: () => Codec[A], filter: Set[A], not: Boolean)    extends Codec[A]
```

Comment:
Instead of `Boolean`, we should introduce:

```scala
sealed trait FilterMode
object FilterMode
  case object Inside extends FilterMode
  case object Outside extends FilterMode
}
```

Then `filter` can use `Inside` and `filterNot` can use `Outside`.

Reply:
Is it to **avoid using booleans** ?

Reply:
It's not performance related, just "best practice" related.

Reply:
I see, thank you!

Reply:
Done in #12